pkg/controller/node.go

```diff
@@ -84,11 +86,11 @@ func (r *ReconcileNode) Reconcile(ctx context.Context, request reconcile.Request
 	// If the node has no DeletionTimestamp, it means it's either create event or update event
 	if n.ObjectMeta.DeletionTimestamp.IsZero() && !n.Spec.Unschedulable {
 		// Add the new schedulable node in availableAttachmentsMap
-		r.addNodeToAvailableAttachmentsMap(ctx, n.Name, n.GetLabels())
-
-		// Node is schedulable, proceed to attempt creation of replica attachment
-		logger.Info("Node is now available. Will requeue failed replica creation requests.")
-		r.tryCreateFailedReplicas(ctx, nodeavailability)
+		if ok := r.addNodeToAvailableAttachmentsMap(ctx, n.Name, n.GetLabels()); ok {
```

Comment:
With the added check, if the node exists in the map, we will return false and will fail to create replicas if the node was uncordonded

Reply:
Exactly, I'll change it to return true if the node is in the map anyway.